app/models/manageiq/providers/foreman/configuration_manager.rb

```diff
@@ -31,11 +37,54 @@ def self.description
     @description ||= "Foreman Configuration".freeze
   end
 
+  def self.create_from_params(params, endpoints, authentications)
+    new(params).tap do |ems|
+      endpoints.each { |endpoint| ems.assign_nested_endpoint(endpoint) }
+      authentications.each { |authentication| ems.assign_nested_authentication(authentication) }
+
+      ems.provider.save!
+      ems.save!
+    end
+  end
+
+  def edit_with_params(params, endpoints, authentications)
+    tap do |ems|
+      transaction do
+        # Remove endpoints/attributes that are not arriving in the arguments above
+        ems.endpoints.where.not(:role => nil).where.not(:role => endpoints.map { |ep| ep['role'] }).delete_all
+        ems.authentications.where.not(:authtype => nil).where.not(:authtype => authentications.map { |au| au['authtype'] }).delete_all
+
+        ems.assign_attributes(params)
+        ems.endpoints = endpoints.map(&method(:assign_nested_endpoint))
+        ems.authentications = authentications.map(&method(:assign_nested_authentication))
+
+        ems.provider.save!
+        ems.save!
+      end
+    end
+  end
```

Comment:
I've been having a lot of issues with getting the provider autosave working across all the different provider repos.  I want to unblock @skateman while we work those out and the simplest way to do that is to manually save the provider while creating/editing the configuration_manager via the API

Reply:
This is ugly, but if it works, I'm good with that.